guardian/support-service-lambdas
# Zuora <-> salesforce link remover - update sf billing accounts

## What does this change?
Create a new lambda that updates billing account in Salesforce

<img width="566" alt="Screenshot 2024-07-04 at 12 20 44" src="https://github.com/guardian/support-service-lambdas/assets/36296660/3aef0f26-64d6-42b0-a232-b9aa76c30a80">

- updates billing accounts via the composite api that allows for multiple records to be updated in one callout. The value for GDPR_Update_Attempt__c on billing account record in Salesforce will be incremented by 1. An alerting system already exists within Salesforce that fires upon 5 failed update attempts.
- lambda will take in a list of billing account objects generated by the state machine (yet to be built, but coming soon), so for now I have mocked this input.

## How to test
- Execute the lambda and verify the records in Salesforce are updated as expected
